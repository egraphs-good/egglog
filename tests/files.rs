use egg_smol::*;

fn test_program(program: &str, message: &str) {
    let mut egraph = EGraph::default();
    match egraph.parse_and_run_program(program) {
        Ok(msgs) => {
            for msg in msgs {
                log::info!("  {}", msg);
            }
        }
        Err(err) => panic!("{}: {err}", message),
    }
}

fn run(path: &str) {
    let _ = env_logger::builder().is_test(true).try_init();
    let program = std::fs::read_to_string(path).unwrap();
    test_program(&program, "Top level error");

    let egraph = EGraph::default();
    let program_str = egraph
        .parse_program(&program)
        .unwrap()
        .into_iter()
        .map(|x| x.to_string())
        .collect::<Vec<String>>()
        .join("\n");
    test_program(
        &program_str,
        &format!(
            "Program:\n{}\n ERROR after parse, to_string, and parse again.",
            program_str
        ),
    );
}

// include the tests generated from the build script
include!(concat!(std::env!("OUT_DIR"), "/files.rs"));
