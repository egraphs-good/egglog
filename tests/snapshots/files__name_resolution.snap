---
source: tests/files.rs
expression: serialized
---
nodes:
  b-0:
    op: b
    children: []
    eclass: "5"
    cost: 9223372036854776000
  c-0:
    op: c
    children: []
    eclass: "5"
    cost: 9223372036854776000
  zero-0:
    op: zero
    children: []
    eclass: "0"
    cost: 9223372036854776000
  i64-0:
    op: "0"
    children: []
    eclass: i64-0
    cost: 0
  Num-0:
    op: Num
    children:
      - i64-0
    eclass: "0"
    cost: 1
  i64-4208978898528913939:
    op: "7"
    children: []
    eclass: i64-4208978898528913939
    cost: 0
  Num-4208978898528913939:
    op: Num
    children:
      - i64-4208978898528913939
    eclass: "3"
    cost: 1
  i64-15952540911656918845:
    op: "9"
    children: []
    eclass: i64-15952540911656918845
    cost: 0
  Num-15952540911656918845:
    op: Num
    children:
      - i64-15952540911656918845
    eclass: "4"
    cost: 1
  i64-1714775736476281168:
    op: "16"
    children: []
    eclass: i64-1714775736476281168
    cost: 0
  Num-1714775736476281168:
    op: Num
    children:
      - i64-1714775736476281168
    eclass: "5"
    cost: 1
  i64-17615343019692007359:
    op: "3"
    children: []
    eclass: i64-17615343019692007359
    cost: 0
  Num-17615343019692007359:
    op: Num
    children:
      - i64-17615343019692007359
    eclass: "2"
    cost: 1
  a-0:
    op: a
    children: []
    eclass: "2"
    cost: 9223372036854776000
  Add-16275225025205966978:
    op: Add
    children:
      - Num-4208978898528913939
      - Num-15952540911656918845
    eclass: "5"
    cost: 1
  Add-11743562013128004906:
    op: Add
    children:
      - Num-0
      - a-0
    eclass: "2"
    cost: 1
root_eclasses: []
class_data:
  "5":
    type: Math
  "0":
    type: Math
  i64-0:
    type: i64
  "3":
    type: Math
  i64-4208978898528913939:
    type: i64
  "4":
    type: Math
  i64-15952540911656918845:
    type: i64
  i64-1714775736476281168:
    type: i64
  "2":
    type: Math
  i64-17615343019692007359:
    type: i64

